### PR TITLE
Assessors applications list

### PR DESCRIPTION
**Pull request by jaleszek**
- applications list for assessors - implemented showing name, reference, assigned assessors
- application view for assessors - collected the views from the admin module, applicable for the assessor
- sketch of form to adding the assessors (primary/secondary) per application - wireframes are not clarified here

### CODE REVIEW

File: app/models/form_answer.rb [hunk 1]
```diff
@@ -53,6 +53,23 @@ class FormAnswer < ActiveRecord::Base
     has_many :support_letters, dependent: :destroy
     has_many :comments, as: :commentable
     has_many :form_answer_transitions
+    has_many :assessor_assignments
+
+    has_many :assessors, through: :assessor_assignments do
+      def primary
+        where(assessor_assignments: {
+          position: AssessorAssignment::PRIMARY_POSITION
```
Comment (matthewford) [inline app/models/form_answer.rb]: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: app/models/form_answer.rb [hunk 2]
```diff
@@ -53,6 +53,23 @@ class FormAnswer < ActiveRecord::Base
...
+          }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Indent the right brace the same as the first position after the preceding left parenthesis.

File: app/models/form_answer.rb [hunk 3]
```diff
@@ -53,6 +53,23 @@ class FormAnswer < ActiveRecord::Base
...
+        ).first
+      end
+
+      def secondary
+        where(assessor_assignments: {
+          position: AssessorAssignment::SECONDARY_POSITION
```
Comment (matthewford) [inline app/models/form_answer.rb]: Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.

File: app/models/form_answer.rb [hunk 4]
```diff
@@ -53,6 +53,23 @@ class FormAnswer < ActiveRecord::Base
...
+          }
```
Comment (matthewford) [inline app/models/form_answer.rb]: Indent the right brace the same as the first position after the preceding left parenthesis.

File: app/models/form_answer.rb [hunk 5]
```diff
@@ -53,6 +53,23 @@ class FormAnswer < ActiveRecord::Base
     has_many :support_letters, dependent: :destroy
     has_many :comments, as: :commentable
     has_many :form_answer_transitions
+    has_many :assessor_assignments
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 6]
```diff
@@ -53,6 +53,23 @@ class FormAnswer < ActiveRecord::Base
...
+
+    has_many :assessors, through: :assessor_assignments do
```
Comment (matthewford) [inline app/models/form_answer.rb]: Inconsistent indentation detected.

File: app/models/form_answer.rb [hunk 7]
```diff
@@ -53,6 +53,21 @@ class FormAnswer < ActiveRecord::Base
...
+      def primary
+        where(assessor_assignments: {
+          position: AssessorAssignment::PRIMARY_POSITION
+        }).first
+      end
+
+      def secondary
+        where(assessor_assignments: {
+          position: AssessorAssignment::SECONDARY_POSITION
+        }).first
```
Comment (matthewford) [inline app/models/form_answer.rb]: Indent the right brace the same as the first position after the preceding left parenthesis.

File: app/models/assessor.rb [hunk 8]
```diff
@@ -37,12 +44,42 @@ def self.roles
...
+
+  def applications_assigned_to_as(roles = ["regular", "lead"])
+    FormAnswer.for_award_type(assigned_categories_as(roles))
+  end
+
+  def self.role_meth(category)
+    "#{category}_role"
+  end
+
+  def full_name
```
Comment (RuslanHamidullin) [inline app/models/assessor.rb]: minor thing: full_name can be moved to decorator

File: app/views/assessor/form_answers/index.html.slim [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+                th
+                  ' Assessor 1
+                th
+                  ' Assessor 2
+              th &nbsp;
+              th.sortable
+                = sort_link f, "Flag", @search, :flag, disabled: @search.query?
+              th &nbsp;
+          tbody
+            - FormAnswerDecorator.decorate_collection(@form_answers).each do |obj|
```
Comment (RuslanHamidullin) [inline app/views/assessor/form_answers/index.html.slim]: minor thing: view is too big, would be great to split on partials

**Comment by RuslanHamidullin** [comment]
Looks good to merge :+1:

**Pull request merged**